### PR TITLE
Add a changelog file

### PR DESCRIPTION
**Pull request by Martin2112**
Built from Github release API results. Unfortunately this doesn't provide the commit date, just the date the release was published.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+## v1.0.2 - TreeGC, Go 1.9, Update Private Keys.
+
+Published 2018-02-05 15:18:40 +0000 UTC
+
+Go 1.9 is required.
+
+It is now possible to update private keys via the admin API and this was added to the available field masks. The key storage format has not changed so we believe this change is transparent.
+
+Deleted trees are now garbage collected after an interval. This hard deletes them and they cannot be recovered. Be aware of this before upgrading if you have any that in a soft deleted state.
```
Comment (daviddrysdale) [inline CHANGELOG.md]: that *are* in a soft deleted state

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+Commit [dec673baf984c3d22d7b314011d809258ec36821](https://api.github.com/repos/google/trillian/commits/dec673baf984c3d22d7b314011d809258ec36821) Download [zip](https://api.github.com/repos/google/trillian/zipball/v1.0.5)
+
+## v1.0.4 - Fix election issue. Large vendor updates.
+
+Published 2018-02-05 15:42:25 +0000 UTC
+
+An issue has been fixed where the master for a log could resign from the election while it was in the process of integrating a batch of leaves. We do not believe this could cause any issues with data integrity because of the versioned tree storage.
+
+This release includes a large number of vendor commits merged to catch up with etcd and GRPC v1.3.
```
Comment (daviddrysdale) [inline CHANGELOG.md]: I think gRPC is the canonical capitalization (passim).

Do we know the etcd version that got vendored in?

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+Commit [b20b3109af7b68227c83c5d930271eaa4f0be771](https://api.github.com/repos/google/trillian/commits/b20b3109af7b68227c83c5d930271eaa4f0be771) Download [zip](https://api.github.com/repos/google/trillian/zipball/v1.0.6)
+
+## v1.0.5 - TLS, Merge Delay Metrics, Easier Admin Tests
+
+Published 2018-02-05 15:54:26 +0000 UTC
+
+The API protos have been rebuilt with GRPC 1.3.
+
+Timestamps have been added to the log leafs in the MySQL database. Before upgrading to this version you **must** make the following schema changes:
```
Comment (daviddrysdale) [inline CHANGELOG.md]: leaves not leafs?

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+Add the following column to the `LeafData` table. If you have existing data in the queue you might have to remove the NOT NULL clause:
+
+`QueueTimestampNanos  BIGINT NOT NULL`
+
+Add the following column to the `SequencedLeafData` table:
+
+`IntegrateTimestampNanos BIGINT NOT NULL`
```
Comment (daviddrysdale) [inline CHANGELOG.md]: Maybe make the 4 lines above into bullets + code blocks to make the grouping clearer:

 - Add the following column to the `LeafData` table. If you have existing data in the queue you might have to remove the `NOT NULL` clause:
```QueueTimestampNanos  BIGINT NOT NULL```
 - Add the following column to the `SequencedLeafData` table:
￼ ```IntegrateTimestampNanos BIGINT NOT NULL```

**Comment by Martin2112** [comment]
Generated by a slightly modified version of https://github.com/aktau/github-release, which I might submit a PR for.

File: CHANGELOG.md (hunk 3 above)
Comment (Martin2112) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 4 above)
Comment (Martin2112) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (Martin2112) [reply CHANGELOG.md]: Done. Etcd was 3.2.0, added.

File: CHANGELOG.md (hunk 1 above)
Comment (Martin2112) [reply CHANGELOG.md]: Done.

**Pull request merged**